### ISSUE TITLE
Can't build frontend service in matchmaker101/frontend tutorial

### ISSUE THREAD

**Issue by scosgrave** [googleforgames/open-match#1171]
**What happened**:
I received the following error while following the matchmaker101 "[Build your own Game Frontend](https://open-match.dev/site/docs/tutorials/matchmaker101/frontend/)" tutorial when I executed the `docker build` command in the "[Build and Push](https://open-match.dev/site/docs/tutorials/matchmaker101/frontend/#build-and-push)" step:

```
❯ docker build -t scosgrave/mm101-tutorial-frontend frontend/
Sending build context to Docker daemon  52.74kB
Step 1/6 : FROM golang:alpine as go
alpine: Pulling from library/golang
aad63a933944: Pull complete
c732a2540651: Pull complete
7b2225181d6b: Pull complete
c8dae7ec6990: Pull complete
08684ee472f3: Pull complete
Digest: sha256:244a736db4a1d2611d257e7403c729663ce2eb08d4628868f9d9ef27354
Status: Downloaded newer image for golang:alpine
 ---> 760fdda71c8f
Step 2/6 : WORKDIR /app
 ---> Running in d626ff2b5ecc
Removing intermediate container d626ff2b5ecc
 ---> f5e457966455
Step 3/6 : ENV GO111MODULE=on
 ---> Running in 0b169b76bd25
Removing intermediate container 0b169b76bd25
 ---> 383a431fdc04
Step 4/6 : COPY . .
 ---> 8a51d70c7dd0
Step 5/6 : RUN go build -o frontend .
 ---> Running in f206fb482092
go: open-match.dev/open-match@v0.0.0-dev: parsing /open-match/go.mod: open no such file or directory
The command '/bin/sh -c go build -o frontend .' returned a non-zero code:
```

**What you expected to happen**:
I expected the docker build command to complete successfully and create a docker image.

**How to reproduce it (as minimally and precisely as possible)**:

1. Install open-match v0.9.0 in a kubernetes cluster using the installation steps [documented here](https://open-match.dev/site/docs/installation/helm/) (standard Helm v3 installation steps)
2. Follow the tutorial for customizing the frontend [here](https://open-match.dev/site/docs/tutorials/matchmaker101/frontend/)
3. You should see the reported error when executing the "Build and Push" step

**Anything else we need to know?**:

**Output of `kubectl version`**:
```
❯ kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.9", GitCommit:"500f5aba80d71253cc01ac6a8622b8377f4a7ef9", GitTreeState:"clean", BuildDate:"2019-11-13T11:21:43Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-25T15:45:25Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```

**Cloud Provider/Platform (AKS, GKE, Minikube etc.)**: 
```
❯ minikube version
minikube version: v1.8.2
commit: eb13446e786c9ef70cb0a9f85a633194e62396a1
```
Running on Windows 10

**Open Match Release Version**:
```
❯ helm3 list
NAME            NAMESPACE       REVISION        UPDATED                                   STATUS          CHART                   APP VERSION
my-om-release   open-match      1               2020-03-24 11:50:39.2988155 -0700 DST     deployed        open-match-0.9.0        0.9.0
```

**Install Method(yaml/helm):**:
Helm

**Comment by yfei1** [comment]
Hey @scosgrave, are you using the master branch for the tutorial? Could you switch to branch v0.9.0 and see if it works?

**Comment by scosgrave** [comment]
@yfei1  - yes, I was using the master branch. I will try the v0.9.0 branch instead.

**Comment by scosgrave** [comment]
The release-0.9 branch works:
```
 ~/d/o/open-match/tu/matchmaker101  release-0.9#v0.9.0 !2 
❯ docker build -t scosgrave/mm101-tutorial-frontend frontend/
Sending build context to Docker daemon  52.74kB
Step 1/6 : FROM golang:alpine as go
 ---> 760fdda71c8f
Step 2/6 : WORKDIR /app
 ---> Using cache
 ---> f5e457966455
Step 3/6 : ENV GO111MODULE=on
 ---> Using cache
 ---> 383a431fdc04
Step 4/6 : COPY . .
 ---> 54647c269c1e
Step 5/6 : RUN go build -o frontend .
 ---> Running in 4d58e4c651b3
go: downloading open-match.dev/open-match v0.9.0
go: downloading google.golang.org/grpc v1.25.0
go: downloading github.com/golang/protobuf v1.3.2
go: downloading github.com/grpc-ecosystem/grpc-gateway v1.12.0
go: downloading google.golang.org/genproto v0.0.0-20191028173616-919d9bdd9fe6
go: downloading golang.org/x/net v0.0.0-20191105084925-a882066a44e0
go: downloading golang.org/x/sys v0.0.0-20191105231009-c1f44814a5cd
go: downloading golang.org/x/text v0.3.2
Removing intermediate container 4d58e4c651b3
 ---> cc437a4f7191
Step 6/6 : CMD ["/app/frontend"]
 ---> Running in 15fcd7fdadb3
Removing intermediate container 15fcd7fdadb3
 ---> 001f61ad8622
Successfully built 001f61ad8622
Successfully tagged scosgrave/mm101-tutorial-frontend:latest
```

**Comment by yfei1** [comment]
Mark as resolved. Please feel free to reopen this issue if you hit it again.

**Comment by cfortuner** [comment]
Hey! I'm still getting this issue on the master branch. Will this be resolved?

go: open-match.dev/open-match@v0.0.0-dev: parsing /go.mod: open /go.mod: no such file or directory

**Comment by yfei1** [comment]
> Hey! I'm still getting this issue on the master branch. Will this be resolved?
> 
> go: open-match.dev/open-match@v0.0.0-dev: parsing /go.mod: open /go.mod: no such file or directory

the master branch is meant for development purpose, could you try running the tutorial on a stable branch instead? (e.g., v1.0.0)

**Comment by Laremere** [comment]
With 1.0, the APIs are stable and the tutorials can now point to a recent version without needing to be perfectly up to date.

**Comment by twaananen** [comment]
Hi, this issue has reappeared for releases 1.1 and 1.2.

**Comment by syntxerror** [comment]
Hi @twaananen. May you provide steps to reproduce?

**Comment by twaananen** [comment]
So I'm following the documents here word to word: https://open-match.dev/site/docs/tutorials/matchmaker101/
I have cloned the open-match repo and switched to release 1.2 branch.
On step1-frontend in the 'Build and Push' section I run this command:
`docker build -t $REGISTRY/mm101-tutorial-frontend frontend/`
And get this error:
...
Step 5/6 : RUN go build -o frontend .
 ---> Running in 7e22984443ee
go: open-match.dev/open-match@v0.0.0-dev (replaced by ../../../): reading /go.mod: open /go.mod: no such file or directory

The problem as I see it is in go.mod file
```
require (
	google.golang.org/grpc v1.36.0
	open-match.dev/open-match v0.0.0-dev
)
replace open-match.dev/open-match v0.0.0-dev => ../../../
```
In version 0.9 where this apparenty worked, that part in go.mod looked like this:
```
require (
	google.golang.org/grpc v1.25.0
	open-match.dev/open-match v0.9.0
)
```
Now I tried changing it to (and just v1.2.0)
```
require (
	google.golang.org/grpc v1.36.0
	open-match.dev/open-match v1.2.0-rc.1
)
```
but with no luck. It then gives the error:
...
Step 5/6 : RUN go build -o frontend .
 ---> Running in 1568604e1fd8
go: open-match.dev/open-match@v1.2.0-rc.1: missing go.sum entry; to add it:
        go mod download open-match.dev/open-match
The command '/bin/sh -c go build -o frontend .' returned a non-zero code: 1

**Comment by twaananen** [comment]
Right, maybe this was just my non-understanding of go, as I haven't worked with it much. But I could get 1.1.0 to build by changing the go.mod file to:
```
require (
	google.golang.org/grpc v1.36.0
	open-match.dev/open-match v1.1.0
)
```
then going to the frontend folder and run
`go mod tidy`
after that the docker build command went through. This didn't work for 1.2.0.

**Comment by syntxerror** [comment]
I am looking into this. I reproduced and can verify what you're experiencing holds up. I believe it may have something to do with it being an release candidate as I have started the release process for `1.2.0` today and have not come across this. I don't expect you to experience this once `1.2.0` is cut but for release candidates it is an issue.

**Comment by dav-otero** [comment]
Hi! I have been having the same problem and fixed it before seeing this post. 
Since release 1.0.0, tutorials for matchmaker101 and 102 are broken due to the same reason. 
In order to fix it in version 1.2.0, I had to change for every component of the tutorial, the `go.mod` file to point to the right version. For example:
```
$cat director/go.mod
module open-match.dev/open-match/tutorials/matchmaker101/director

go 1.14

require (
	google.golang.org/grpc v1.36.0
	open-match.dev/open-match v1.2.0
)
```
Aside from this, you have to go into the corresponding component and run:
```
go mod tidy
```
to update `go.sum` file.

For some reason though, using the `main` branch fails with the error:
```
go: open-match.dev/open-match@v0.0.0-dev: parsing /open-match/go.mod: open no such file or directory
```
that is related to the `go.mod` files not pointing to a `proper` version, and the replacement seems to not be ok:

```
require (
	google.golang.org/grpc v1.36.0
	open-match.dev/open-match v0.0.0-dev
)
replace open-match.dev/open-match v0.0.0-dev => ../../../
```

Note that `v1.2.0` has the `go.mod file looking like this`, probably due to a missing change in the process of releasing 1.2.0.

**Comment by syntxerror** [comment]
I have made a change for any running v1.1 originating from this issue on PR #1400. For those who are running v1.1.0, please use the v1.1.1. Release notes to follow shortly.
